builder/frameworks/espidf.py

```
@@ -1099,21 +1118,45 @@ def run_cmake(src_dir, build_dir, extra_args=None):
+    This ensures consistency with the Arduino framework's lib_ignore handling.
+    """
+    try:
+        # Create a LibraryIgnoreHandler instance to process lib_ignore
+        config = _component_manager.ComponentManagerConfig(env)
+        logger = _component_manager.ComponentLogger()
+        lib_handler = _component_manager.LibraryIgnoreHandler(config, logger)
+        
+        # Get the processed lib_ignore entries (already converted to component names)
+        lib_ignore_entries = lib_handler._get_lib_ignore_entries()
+        
+        return lib_ignore_entries
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)